sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -4721,13 +4721,91 @@ def repeat(self, repeats):
       pd.core.strings.StringMethods, 'get_dummies',
       reason='non-deferred-columns')
 
-  split = frame_base.wont_implement_method(
-      pd.core.strings.StringMethods, 'split',
-      reason='non-deferred-columns')
+  def _split_helper(
+    self, rsplit=False, pat=None, expand=False, regex=None, **kwargs):
+
+    # Adding arguments to kwargs. regex introduced in pandas 1.4
+    # but only for split, not rsplit
+    kwargs['pat'] = pat
+    kwargs['expand'] = expand
+    if PD_VERSION >= (1, 4) and not rsplit:
+      kwargs['regex'] = regex
+
+    if not expand:
+      # Not creating separate columns
+      proxy = self._expr.proxy()
+      func = lambda s: pd.concat([proxy,
+        (s.str.split(**kwargs) if not rsplit else s.str.rsplit(**kwargs))]
```

Comment:
could you branch on `rsplit` outside of the function definition?

Reply:
Yup, changed!